backend/apps/github/common.py

```diff
@@ -39,164 +39,156 @@ def sync_repository(
     entity_key = gh_repository.name.lower()
     is_owasp_site_repository = check_owasp_site_repository(entity_key)
 
-    # GitHub repository organization.
-    if organization is None:
-        gh_organization = gh_repository.organization
-        if gh_organization is not None:
-            organization = Organization.update_data(gh_organization)
-
-    # GitHub repository owner.
-    if user is None:
-        user = User.update_data(gh_repository.owner)
-
-    # GitHub repository.
-    commits = gh_repository.get_commits()
-    contributors = gh_repository.get_contributors()
-    languages = None if is_owasp_site_repository else gh_repository.get_languages()
+    # Setup organization and user
+    organization = organization or (
+        Organization.update_data(gh_repository.organization)
+        if gh_repository.organization
+        else None
+    )
+    user = user or User.update_data(gh_repository.owner)
 
+    # Create repository
     repository = Repository.update_data(
         gh_repository,
-        commits=commits,
-        contributors=contributors,
-        languages=languages,
+        commits=gh_repository.get_commits(),
+        contributors=gh_repository.get_contributors(),
+        languages=None if is_owasp_site_repository else gh_repository.get_languages(),
         organization=organization,
         user=user,
     )
 
+    # Process repository content if not archived
     if not repository.is_archived:
-        # GitHub repository milestones.
-        kwargs = {
-            "direction": "desc",
-            "sort": "updated",
-            "state": "all",
-        }
-
-        until = (
-            latest_updated_milestone.updated_at
-            if (latest_updated_milestone := repository.latest_updated_milestone)
-            else timezone.now() - td(days=30)
+        _sync_repository_milestones(gh_repository, repository)
+        _sync_repository_issues(gh_repository, repository)
+        _sync_repository_pull_requests(gh_repository, repository)
+
+    _sync_repository_releases(gh_repository, repository, is_owasp_site_repository)
+    _sync_repository_contributors(gh_repository, repository)
```

Comment:
It looks like these can be defined as `Repository` model methods, it's fine to make it fatter.